src/script/services/manifest.ts

```diff
@@ -144,21 +165,49 @@ export async function fetchManifest(
     const promiseAnyOrPolyfill: (
       promises: Promise<ManifestDetectionResult>[]
     ) => Promise<ManifestDetectionResult> = promises =>
-        Promise['any'] ? Promise['any'](promises) : promiseAnyPolyfill(promises);
+      Promise['any'] ? Promise['any'](promises) : promiseAnyPolyfill(promises);
 
     try {
       testResult = await promiseAnyOrPolyfill(manifestDetectors);
 
-      manifest = testResult.content;
+      if (!fetchAttempted) {
+        manifest = testResult.content;
+
+        if (!manifest.icons) {
+          manifest.icons = [];
+        }
+
+        if (!manifest.screenshots) {
+          manifest.screenshots = [];
+        }
```

Comment:
we had logic to guard against the arrays not being initialized, hence this inclusion